ckanext/xloader/helpers.py

```diff
@@ -25,3 +26,17 @@ def xloader_status_description(status):
         return captions.get(status['status'], status['status'].capitalize())
     else:
         return _('Not Uploaded Yet')
+
+
+def is_resource_supported_by_xloader(res_dict, check_access = True):
+    is_supported_format = XLoaderFormats.is_it_an_xloader_format(res_dict.get('format'))
+    is_datastore_active = res_dict.get('datastore_active', False)
+    if check_access:
+        user_has_access = toolkit.h.check_access('package_update', {'id':res_dict.get('package_id')})
+    else:
+        user_has_access = True
+    try:
+        is_supported_url_type = res_dict.get('url_type') not in toolkit.h.datastore_rw_resource_url_types()
+    except AttributeError:
+        is_supported_url_type = (res_dict.get('url_type') == 'upload' or res_dict.get('url_type') == '')
```

Comment:
This should handle any falsy url_type the same way as empty string. It is possible for url_type to be `None`.

Reply:
changed to falsy check instead of just blank string.